src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -94,13 +94,10 @@
            new-hash      (calculate-table-hash db-metadata)
            hash-changed? (or (not old-hash) (not= new-hash old-hash))]
        ;; if hash is unchanged we can skip the rest of the sync process
-       (when-not hash-changed?
```

Comment:
Unless I'm missing something, it's safe to replace these two disjunct `when-not`s with an if.